commit bffa91b48d39779aeed55225ce2ca211c8a626e0
Author: Nikhil Bhatia <116085493+nikhilbhatia08@users.noreply.github.com>
Date:   Thu Oct 2 12:06:43 2025 +0530

    [TRACE SDK] Batch span processor options now using env variables (#3661)

diff --git b/sdk/include/opentelemetry/sdk/trace/batch_span_processor_options.h a/sdk/include/opentelemetry/sdk/trace/batch_span_processor_options.h
index 32c0127a..ac105238 100644
--- b/sdk/include/opentelemetry/sdk/trace/batch_span_processor_options.h
+++ a/sdk/include/opentelemetry/sdk/trace/batch_span_processor_options.h
@@ -17,30 +17,22 @@ namespace trace
 /**
  * Struct to hold batch SpanProcessor options.
  */
-struct OPENTELEMETRY_EXPORT BatchSpanProcessorOptions
+struct BatchSpanProcessorOptions
 {
-  BatchSpanProcessorOptions();
   /**
    * The maximum buffer/queue size. After the size is reached, spans are
    * dropped.
    */
-  size_t max_queue_size;
+  size_t max_queue_size = 2048;
 
   /* The time interval between two consecutive exports. */
-  std::chrono::milliseconds schedule_delay_millis;
-
-  /**
-   * The maximum time allowed to to export data
-   * It is not currently used by the SDK and the parameter is ignored
-   * TODO: Implement the parameter in BatchSpanProcessor
-   */
-  std::chrono::milliseconds export_timeout;
+  std::chrono::milliseconds schedule_delay_millis = std::chrono::milliseconds(5000);
 
   /**
    * The maximum batch size of every export. It must be smaller or
    * equal to max_queue_size.
    */
-  size_t max_export_batch_size;
+  size_t max_export_batch_size = 512;
 };
 
 }  // namespace trace
diff --git b/sdk/src/trace/CMakeLists.txt a/sdk/src/trace/CMakeLists.txt
index 95eb541e..5d7a36db 100644
--- b/sdk/src/trace/CMakeLists.txt
+++ a/sdk/src/trace/CMakeLists.txt
@@ -13,7 +13,6 @@ add_library(
   provider.cc
   batch_span_processor.cc
   batch_span_processor_factory.cc
-  batch_span_processor_options.cc
   simple_processor_factory.cc
   samplers/always_on_factory.cc
   samplers/always_off_factory.cc
diff --git b/sdk/test/trace/batch_span_processor_test.cc a/sdk/test/trace/batch_span_processor_test.cc
index 1e8d5734..af9bbeab 100644
--- b/sdk/test/trace/batch_span_processor_test.cc
+++ a/sdk/test/trace/batch_span_processor_test.cc
@@ -2,7 +2,6 @@
 // SPDX-License-Identifier: Apache-2.0
 
 #include <gtest/gtest.h>
-#include <stdlib.h>
 #include <algorithm>
 #include <atomic>
 #include <chrono>
@@ -26,12 +25,6 @@
 #include "opentelemetry/sdk/trace/span_data.h"
 #include "opentelemetry/version.h"
 
-#if defined(_MSC_VER)
-#  include "opentelemetry/sdk/common/env_variables.h"
-using opentelemetry::sdk::common::setenv;
-using opentelemetry::sdk::common::unsetenv;
-#endif
-
 OPENTELEMETRY_BEGIN_NAMESPACE
 
 /**
@@ -371,78 +364,4 @@ TEST_F(BatchSpanProcessorTestPeer, TestScheduleDelayMillis)
   }
 }
 
-TEST(BatchSpanProcessorOptionsEnvTest, TestDefaultValues)
-{
-  sdk::trace::BatchSpanProcessorOptions options;
-
-  EXPECT_EQ(options.max_queue_size, static_cast<size_t>(2084));
-  EXPECT_EQ(options.schedule_delay_millis, std::chrono::milliseconds(5000));
-  EXPECT_EQ(options.export_timeout, std::chrono::milliseconds(3000));
-  EXPECT_EQ(options.max_export_batch_size, static_cast<size_t>(512));
-}
-
-TEST(BatchSpanProcessorOptionsEnvTest, TestMaxQueueSizeFromEnv)
-{
-  setenv("OTEL_BSP_MAX_QUEUE_SIZE", "1234", 1);
-
-  sdk::trace::BatchSpanProcessorOptions options;
-
-  EXPECT_EQ(options.max_queue_size, static_cast<size_t>(1234));
-
-  unsetenv("OTEL_BSP_MAX_QUEUE_SIZE");
-}
-
-TEST(BatchSpanProcessorOptionsEnvTest, TestScheduleDelayFromEnv)
-{
-  setenv("OTEL_BSP_SCHEDULE_DELAY", "7s", 1);
-
-  sdk::trace::BatchSpanProcessorOptions options;
-
-  EXPECT_EQ(options.schedule_delay_millis, std::chrono::milliseconds(7000));
-
-  unsetenv("OTEL_BSP_SCHEDULE_DELAY");
-}
-
-TEST(BatchSpanProcessorOptionsEnvTest, TestExportTimeoutFromEnv)
-{
-  setenv("OTEL_BSP_EXPORT_TIMEOUT", "250ms", 1);
-
-  sdk::trace::BatchSpanProcessorOptions options;
-
-  EXPECT_EQ(options.export_timeout, std::chrono::milliseconds(250));
-
-  unsetenv("OTEL_BSP_EXPORT_TIMEOUT");
-}
-
-TEST(BatchSpanProcessorOptionsEnvTest, TestMaxExportBatchSizeFromEnv)
-{
-  setenv("OTEL_BSP_MAX_EXPORT_BATCH_SIZE", "42", 1);
-
-  sdk::trace::BatchSpanProcessorOptions options;
-
-  EXPECT_EQ(options.max_export_batch_size, static_cast<size_t>(42));
-
-  unsetenv("OTEL_BSP_MAX_EXPORT_BATCH_SIZE");
-}
-
-TEST(BatchSpanProcessorOptionsEnvTest, TestOptionsReadFromMultipleEnvVars)
-{
-  setenv("OTEL_BSP_MAX_QUEUE_SIZE", "3000", 1);
-  setenv("OTEL_BSP_SCHEDULE_DELAY", "2s", 1);
-  setenv("OTEL_BSP_EXPORT_TIMEOUT", "1s", 1);
-  setenv("OTEL_BSP_MAX_EXPORT_BATCH_SIZE", "256", 1);
-
-  sdk::trace::BatchSpanProcessorOptions options;
-
-  EXPECT_EQ(options.max_queue_size, static_cast<size_t>(3000));
-  EXPECT_EQ(options.schedule_delay_millis, std::chrono::milliseconds(2000));
-  EXPECT_EQ(options.export_timeout, std::chrono::milliseconds(1000));
-  EXPECT_EQ(options.max_export_batch_size, static_cast<size_t>(256));
-
-  unsetenv("OTEL_BSP_MAX_QUEUE_SIZE");
-  unsetenv("OTEL_BSP_SCHEDULE_DELAY");
-  unsetenv("OTEL_BSP_EXPORT_TIMEOUT");
-  unsetenv("OTEL_BSP_MAX_EXPORT_BATCH_SIZE");
-}
-
 OPENTELEMETRY_END_NAMESPACE
